### PR TITLE
Fixed the #define ENABLE_MQTT_TLS issue for libraries, other error fixes

### PR DESCRIPTION
**Pull request by quinnmiller1997**
Now has issue with defining ENABLE_MQTT_TLS in the libraries regardless if it is defined by the user in the sketch.  The WolfSSL library is not included in the sketch when ENABLE_MQTT_TLS is not defined, the libraries throw an error because they try and include files from it

### CODE REVIEW

File: IDE/ARDUINO/wolfmqtt_client/wolfmqtt_client.ino [hunk 1]
```diff
@@ -3,7 +3,9 @@
 //#define ENABLE_MQTT_TLS
 
 #ifdef ENABLE_MQTT_TLS
-#include <config.h>
+#ifdef HAVE_CONFIG
+  #include <config.h>
+#endif
```
Comment (dgarske) [inline IDE/ARDUINO/wolfmqtt_client/wolfmqtt_client.ino]: I believe with the change to `mqtt_types.h` for `ENABLE_MQTT_TLS` this section from line 1-8 can be removed. Instead please add a comment similar to this into `mqtt_types.h` and comment out `ENABLE_MQTT_TLS ` by default. Also please add a comment in the `README.md` about enabling TLS. Please include a link to the wolfSSL IDE/ARDUINO project. Thanks!

File: IDE/ARDUINO/wolfmqtt_client/wolfmqtt_client.ino [hunk 2]
```diff
@@ -218,6 +222,7 @@ void loop() {
   rc = MqttClient_NetConnect(&client, mHost, mPort,
                              DEFAULT_CON_TIMEOUT_MS, use_tls,
 #ifdef ENABLE_MQTT_TLS
+
```
Comment (dgarske) [inline IDE/ARDUINO/wolfmqtt_client/wolfmqtt_client.ino]: Please remove this extra newline.

File: IDE/ARDUINO/wolfmqtt_client/wolfmqtt_client.ino [hunk 3]
```diff
@@ -1,13 +1,5 @@
-/* Uncomment this to enable TLS support */
-/* Make sure and include the wolfSSL library */
-//#define ENABLE_MQTT_TLS
-
-#ifdef ENABLE_MQTT_TLS
-#include <config.h>
 #include <wolfssl.h>
```
Comment (dgarske) [inline IDE/ARDUINO/wolfmqtt_client/wolfmqtt_client.ino]: Please also remove these two wolfssl references.

File: wolfmqtt/mqtt_types.h [hunk 4]
```diff
@@ -27,10 +27,16 @@
...
+/* configuration for Arduino */
+#ifdef ARDUINO
 /* Uncomment this to enable TLS support */
 /* Make sure and include the wolfSSL library */
-#ifdef ARDUINO
     //#define ENABLE_MQTT_TLS
+
+    /* make sure arduino can see the wolfssl library directory */
+    #ifdef ENABLE_MQTT_TLS
+        #include <wolfssl.h>
```
Comment (quinnmiller1997) [inline wolfmqtt/mqtt_types.h]: if this isn't included at any point, Arduino can't see the wolfssl library when `#include <wolfssl/__________>' is called

Comment (dgarske) [reply wolfmqtt/mqtt_types.h]: You are correct. The state of this looks good! I'll give it one test on my own using Arduino and should be able to approve/merge this.

File: wolfmqtt/mqtt_types.h [hunk 5]
```diff
@@ -27,6 +27,18 @@
...
 #define WOLFMQTT_TYPES_H
 
+/* configuration for Arduino */
+#ifdef ARDUINO
+/* Uncomment this to enable TLS support */
+/* Make sure and include the wolfSSL library */
+    //#define ENABLE_MQTT_TLS
+
+    /* make sure arduino can see the wolfssl library directory */
+    #ifdef ENABLE_MQTT_TLS
```
Comment (denravonska) [inline wolfmqtt/mqtt_types.h]: Shouldn't this apply to all platforms which have defined `ENABLE_MQTT_TLS`?

Comment (dgarske) [reply wolfmqtt/mqtt_types.h]: Hi denravonska, thanks for the question. The answer is no. The `#include <wolfssl.h>` is a generated file for Arduino using the `<wolfssl-root>/IDE/ARDUINO/wolfssl-arduino.sh` script. The normal `#include <wolfssl/wolfcrypt/settings.h>` and `#include <wolfssl/ssl.h>` files are included here: https://github.com/wolfSSL/wolfMQTT/blob/master/wolfmqtt/mqtt_socket.h#L35

Comment (denravonska) [reply wolfmqtt/mqtt_types.h]: Ah, right. Thanks.

**Pull request merged**